hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/WALEntryStream.java

```diff
@@ -252,6 +252,7 @@ private void dequeueCurrentLog() throws IOException {
     LOG.debug("EOF, closing {}", currentPath);
     closeReader();
     logQueue.remove();
+    setCurrentPath(null);
```

Comment:
Nice fix, can you add some detail to the commit message, I think it's subtle.

Looks like the issue is if openNextLog() throws, we want currentPath to be null rather than stale.

Reply:
I don't have good context here, but is it possible to add a small test? That could help anyone get to know this bit better, it's fine though if it is too complicated to add it.

Reply:
Got to know from @sandeepvinayak that writing test is complicated. We can skip it.